### PR TITLE
DOC: Add missing changelogs for NEP 52 PRs

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @rgommers,

This PR contains two commits: One with release notes that I obtained by running:
```
towncrier build --version "2.0.0"
```
And second one with manually added release notes for four merged Python API cleanup PRs: https://github.com/numpy/numpy/pull/23830, https://github.com/numpy/numpy/pull/24154, https://github.com/numpy/numpy/pull/24316, https://github.com/numpy/numpy/pull/24357.

### CODE REVIEW

File: doc/source/release/2.0.0-notes.rst [hunk 1]
```diff
@@ -1,45 +1,313 @@
...
+
+* ``np.source`` has been removed. The preferred replacement is 
+  ``inspect.getsource``.
+
+* ``np.lookfor`` has been removed.
+
+  (`gh-24144 <https://github.com/numpy/numpy/pull/24144>`__)
+
+* ``numpy.who`` has been removed. As an alternative for the removed functionality, one
+can use a variable explorer that is available in IDEs such as Spyder or Jupyter Notebook.
```
Comment (charris) [inline doc/source/release/2.0.0-notes.rst:38]: Indentation is off.

File: doc/source/release/2.0.0-notes.rst [hunk 2]
```diff
@@ -1,45 +1,313 @@
...
+* The deprecated function ``fastCopyAndTranspose`` and its C counterpart
+  is now removed.
+* The deprecation of ``PyArray_ScalarFromObject`` is now finalized.
+
+  (`gh-24312 <https://github.com/numpy/numpy/pull/24312>`__)
 
-Compatibility notes
-===================
+* ``np.msort`` has been removed. For a replacement, ``np.sort(a, axis=0)`` 
+should be used instead.
```
Comment (charris) [inline doc/source/release/2.0.0-notes.rst:104]: Indentation is off.

File: doc/source/release/2.0.0-notes.rst [hunk 3]
```diff
@@ -1,45 +1,313 @@
...
-==========================
-
-
-Highlights
-==========
+=========================
 
 
 New functions
 =============
```
Comment (charris) [inline doc/source/release/2.0.0-notes.rst:7]: Indentation.

File: doc/source/release/2.0.0-notes.rst (hunk 1 above)
Comment (mtsokol) [reply doc/source/release/2.0.0-notes.rst]: This part came generated from `towncrier` - Fixed it!

File: doc/source/release/2.0.0-notes.rst [hunk 4]
```diff
@@ -1,45 +1,313 @@
-.. currentmodule:: numpy
```
[the remark this answers is not in the document]
Comment (mtsokol) [reply doc/source/release/2.0.0-notes.rst:1]: Restored!

File: doc/source/release/2.0.0-notes.rst (hunk 1 above)
Comment (charris) [reply doc/source/release/2.0.0-notes.rst]: It probably was in the fragment. That is not unusual :)

File: doc/source/release/2.0.0-notes.rst (hunk 3 above)
Comment (mtsokol) [reply doc/source/release/2.0.0-notes.rst:9]: What should be the desired indentation here? 
I see that e.g. `1.23.0-notes.rst` also has `New functions` written this way.

Comment (charris) [reply doc/source/release/2.0.0-notes.rst:9]: Happens, the notes get copied one to another :) Probably doesn't matter, but headings flush on left works.

**Pull request merged**